extensions/controllers/sandboxclaim_controller.go

```diff
@@ -311,6 +327,31 @@ func (r *SandboxClaimReconciler) getOrCreateSandbox(ctx context.Context, claim *
 			logger.Error(err, "Sandbox controller mismatch")
 			return nil, err
 		}
+
+		needsUpdate := false
+
+		claimTime := claim.Spec.ShutdownTime
+		sandboxTime := sandbox.Spec.ShutdownTime
+
+		if claimTime == nil && sandboxTime != nil {
+			// Claim wants infinity, Sandbox has a limit -> UPDATE
```

Comment:
This comparison logic could be simplified. Instead of three separate checks, you can use `!equality.Semantic.DeepEqual(claimTime, sandboxTime)`. This is more concise and less error-prone.